scripts/check_links.py

```diff
@@ -0,0 +1,48 @@
+#! /usr/bin/env python
+"""
+Goes through all the inline-links in markdown files and reports the breakages.
+"""
+import re
+import sys
+import pathlib
+from multiprocessing.dummy import Pool
+from collections import namedtuple
+from typing import Tuple
+import requests
+
+MatchTuple = namedtuple('MatchTuple', 'source name link')
+
+def url_ok(match_tuple: MatchTuple) -> Tuple[MatchTuple, bool]:
+    try:
+        return (match_tuple, requests.get(match_tuple.link).ok)
+    except requests.ConnectionError:
+        return (match_tuple, False)
+
+if __name__ == "__main__":
+
+    project_root = (pathlib.Path(__file__).parent / "..").resolve() # pylint: disable=no-member
+    markdown_files = project_root.glob('**/*.md')
+
+    all_matches = set()
+    url_regex = re.compile(r'\[([^!][^\]]+)\]\((http[s]?[^)(]+)\)')
+    for markdown_file in markdown_files:
+        with open(markdown_file) as handle:
+            for line in handle.readlines():
+                matches = url_regex.findall(line)
+                for match in matches:
+                    if 'localhost' not in match[1]:
+                        all_matches.add(MatchTuple(source=markdown_file, name=match[0], link=match[1]))
+
+    with Pool(processes=10) as pool:
```

Comment:
How slow is this without multiprocessing? I'd probably prefer without, unless the speedup is significant.

Reply:
yeah its' about 5X. With parallel about 8-12 seconds. Without parallel 40-60 seconds